### PR TITLE
changed select to creatableselect for classification 

### PR DESCRIPTION
**Pull request by sreevardhanreddi**
#53

### CODE REVIEW

**Comment by seveibar** [comment]
Good start, but doesn't propagate changes upward to make the controller know that a new classification has been created

**Comment by sreevardhanreddi** [comment]
> propagate new classifications upward, should have prop on `<Annotator />`

@seveibar I did not understand, could you please elaborate on that, do you mean that I need to add a prop or function on **Annotator** component which allows to add new classification to **regionClsList**? and that prop needs to be sent to the **RegionLabel** component? so whenever a new classification is added it is added to the **regionClsList**

**Comment by seveibar** [comment]
The `<Annotator />` should have a prop like `onRegionLabelAdded`, whatever is managing the state of the `Annotator` can then add the label to the `regionClsList`

I appreciate the work so far btw :)

**Comment by sreevardhanreddi** [comment]
Hi @seveibar , I have updated the code to reflect new options added to **regionClsList**,  have a look at it and let me know if any other changes are required 🙂.

**Comment by seveibar** [comment]
nice implementation! looks good at first glance, going to do some testing :)

File: src/Annotator/index.js [hunk 1]
```diff
@@ -121,6 +121,13 @@ export const Annotator = ({
     dispatchToReducer(action)
   })
 
+  const onRegionLabelAdded = (cls) => {
```
Comment (seveibar) [inline src/Annotator/index.js:124]: use `useEventCallback` here, it'll prevent rerenders

File: src/Annotator/reducers/history-handler.js [hunk 2]
```diff
@@ -26,6 +26,15 @@ export default (reducer) => {
...
     const nextState = reducer(state, action)
 
+    if (action.type === "ON_CLS_ADDED" && action.cls && action.cls !== "") {
+      const oldRegionClsList = state.regionClsList
+      const newState = {
+        ...state,
+        regionClsList: oldRegionClsList.concat(action.cls),
+      }
+      return newState
+    }
```
Comment (seveibar) [inline src/Annotator/reducers/history-handler.js:36]: this should be moved to `general-reducer.js` (since it's not managing history)

**Comment by sreevardhanreddi** [comment]
@seveibar, I have updated the code :)

File: src/Annotator/index.js [hunk 3]
```diff
@@ -132,6 +132,13 @@ export const Annotator = ({
     dispatchToReducer(action)
   })
 
+  const onRegionClassAdded = useEventCallback((cls) => {
+    dispatchToReducer({
+      type: "ON_CLS_ADDED",
```
Comment (puskuruk) [inline src/Annotator/index.js:137]: That's good :dancers:

File: src/Annotator/index.js [hunk 4]
```diff
@@ -132,6 +132,13 @@ export const Annotator = ({
...
+      cls: cls,
```
Comment (puskuruk) [inline src/Annotator/index.js:138]: you could use only "cls" for that. :)

File: src/Annotator/reducers/general-reducer.js [hunk 5]
```diff
@@ -13,6 +13,15 @@ import convertExpandingLineToPolygon from "./convert-expanding-line-to-polygon"
 const getRandomId = () => Math.random().toString().split(".")[1]
 
 export default (state: MainLayoutState, action: Action) => {
+  if (action.type === "ON_CLS_ADDED" && action.cls && action.cls !== "") {
+    const oldRegionClsList = state.regionClsList
+    const newState = {
+      ...state,
+      regionClsList: oldRegionClsList.concat(action.cls),
```
Comment (puskuruk) [inline src/Annotator/reducers/general-reducer.js:20]: That's so clear and understandable :dancers:

File: src/Annotator/index.js (hunk 3 above)
Comment (sreevardhanreddi) [reply src/Annotator/index.js:137]: Thanks!!!

File: src/Annotator/index.js (hunk 4 above)
Comment (sreevardhanreddi) [reply src/Annotator/index.js:138]: yes, I could have

File: src/Annotator/reducers/general-reducer.js (hunk 5 above)
Comment (sreevardhanreddi) [reply src/Annotator/reducers/general-reducer.js:20]: Thanks a lot @puskuruk

**Pull request merged**